osgar/drivers/canserial.py

```diff
@@ -243,12 +243,18 @@ def process_gen(self, data):
     def slot_raw(self, timestamp, data):
         if len(data) > 0:
             for packet in self.process_gen(data):
-                self.bus.publish('can', packet)
+                msg_id, rtr, size = parse_header(packet)
+                if rtr == 0:
+                    assert size + 2 == len(packet), (size, len(packet))
+                # TODO verify how rtr is handled on PCAN?
+                self.bus.publish('can', [msg_id, packet[2:], 0])
 
     def slot_can(self, timestamp, data):
         if self.can_bridge_initialized:
-            # at the moment is can serial just forwarding raw packets
-            self.bus.publish('raw', data)
+            # at the moment is can serial just forwarding translation of raw packets
+            msg_id, payload, flags = data
+            assert flags == 0, flags  # CAN bridge supports only basic addressing
+            self.bus.publish('raw', CAN_packet(msg_id, list(payload)))
```

Comment:
Is this `list`-hack related to msgpack? I think `bytes` roundtrip just fine:
```python
>>> msgpack.unpackb(msgpack.packb(b"hi"))
b'hi'
```
Can we keep using `bytes` as payload?

Reply:
I see. We are using
https://github.com/robotika/osgar/blob/44c581214bf7e455cba62cc3e1fa322a9b1edf7a/osgar/lib/serialize.py#L11-L12
so `bytes` come out as `str`.
```python
>>> type(msgpack.unpackb(msgpack.packb(b'hi')))
<class 'bytes'>
>>> type(msgpack.unpackb(msgpack.packb(b'hi'), raw=False))
<class 'str'>
```
Hmm. I think I'd rather be using `bytes` everywhere. @m3d what is your take on this?

Reply:
well, this particular problem was related to implementation of `CAN_packet()`, not `msgpack`. I believe it could be resolved better - `list` was just quick hack to get it working (i.e. conversion to older code which was using arrays of numbers, not `bytes`).

Regarding `raw=False` I am not sure now, why we picked this option ... could be sending strings:
```
>>> msgpack.packb('hi')
b'\xa2hi'
>>> msgpack.unpackb(msgpack.packb('hi'))
b'hi'
```

Reply:
What about the `list(payload)` stuff? Maybe `CAN_packet` could accept either list of `int` or `bytes()`? It makes bytes from it anyway. We don't have to - I'd just like to make a resolution and not just let it hang.